File: JS/CoinChangeII.test.js
/**
 * @version JAVASCRIPT ECMAScript 6 
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
import TEST from 'tape'
import { change } from './CoinChangeII.js'

TEST('Starting CoinChangeII test...', (t) => {
    //Test 1
    let want = 4
    let got = change( 5, [1,2,5] )
    t.assert( want.toString === got.toString, "Expect: "+want)
    //Test 2
    want = 0
    got = change( 3, [2] )
    t.assert( want.toString === got.toString, "Expect: "+want)
    //Test 3
    want = 1
    got = change( 10, [10] )
    t.assert( want.toString === got.toString, "Expect: "+want)

    t.end()
})

File: JAVA/CoinChangeIITest.java
/** 
 * @version JAVA
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertEquals;
import org.junit.Test;

public class CoinChangeIITest {

    @Test
	public void testCoinChangeII() throws Exception {
        //Test 1
        int want = 4;
        int got =  CoinChangeII.change( 5, new int[]{1,2,5} );
        assertEquals(want, got);
        //Test 2
        want = 0;
        got =  CoinChangeII.change( 3, new int[]{2} );
        assertEquals(want, got);
        //Test 3
        want = 1;
        got =  CoinChangeII.change( 10, new int[]{10} );
        assertEquals(want, got);
	}

}
